crates/otspec/src/layout/common.rs

```diff
@@ -0,0 +1,215 @@
+use crate::{Serialize, Serializer};
+use bitflags::bitflags;
+use otspec::layout::anchor::Anchor;
+use otspec::types::*;
+use otspec::Deserializer;
+use otspec_macros::{tables, Deserialize, Serialize};
+use std::fmt::Debug;
+
+tables!(
```

Comment:
Kind of. It turns these spec cut-and-paste structures into Rust structs with appropriate derives, plus the serde-style annotations.

Reply:
Okay, makes sense.

With your blessing, I'd like to look into just using `derive` directly, and perhaps replacing `Serialize` and `Deserialize` with a single `OtSpec` (or something) macro?

I can also then look into implementing a version that works with `enum`, for handling things like `FeatureParams`?  does that make sense?

Reply:
I'm happy for you to try, but I can't think about it right now. :-) One thing you should note is that often the serialization can be derived automatically but because of data dependencies deserialisation needs to be handled manually. So a single macro may not be possible. See `ValueRecord` for an extreme example.